python/paddle/optimizer/momentum.py

```diff
@@ -277,6 +285,130 @@ def _create_regularization_of_grad(self, param, grad, regularization=None):
+                    else:
+                        self.master_weight_dict['FP16_LODTensor'] = None
+                    # regularization
+                    regularization_method = self._regularization_method
+                    regularization_coeff = self._regularization_coeff
+                    if hasattr(param, 'regularizer'):
+                        # we skip param's l2decay before, so fuse it with momentum here.
+                        if isinstance(param.regularizer, L2DecayRegularizer):
+                            regularization_method = "l2_decay"
+                            regularization_coeff = param.regularizer._regularization_coeff
+                        # the param's regularization has been done before, we avoid do l2decay in momentum.
+                        else:
+                            regularization_method = ""
+                            regularization_coeff = 0
+                    self.regularization_method_dict['FP16_LODTensor'].append(
+                        regularization_method)
+                    self.regularization_coeff_dict['FP16_LODTensor'].append(
+                        regularization_coeff)
+                else:
+                    raise ValueError(
+                        "Now multi_tensor_momentum only support fp32 and fp16 parameters and grad is LOD_TENSOR."
+                    )
+
+    def _create_optimization_pass(self, parameters_and_grads):
+        """Add optimization operators to update gradients to tensors.
+
+        Args:
+          parameters_and_grads(list(tuple(Tensor, Tensor))):
+            a list of (tensor, gradient) pair to update.
+
+        Returns:
+          return_op_list: a list of operators that will complete one step of
+            optimization. This will include parameter update ops, global step
+            update ops and any other custom ops required by subclasses to manage
+            their internal state.
+        """
+        # This is a Implemented create_optimization_pass of paddle.optimizer.
+        # If use_multi_tensor is True, it will call merged_momentum to update
+        # parameters, else it will use default implementation of create_optimization_pass.
+
+        if self._use_multi_tensor:
+            global_block = framework.default_main_program().global_block()
+            target_block = global_block
+            current_block = framework.default_main_program().current_block()
+            if current_block.idx != global_block.idx:
+                assert current_block.backward_block_idx != -1, \
+                    "current block is not global_block, but it doesn't have backward block."
+                target_block = framework.default_main_program().blocks[
+                    current_block.backward_block_idx]
+
+            start = len(target_block.ops)
+            self.helper = LayerHelper(self.__class__.__name__)
+
+            self._create_global_learning_rate()
+            if framework.in_dygraph_mode():
```

Comment:
what about static mode?

Reply:
Tks, Multi Tensor has been added to the static mode.